modules/juce_audio_devices/native/juce_android_Midi.cpp

```diff
@@ -373,23 +373,27 @@ class MidiChangeDetector
 public:
     void addListener(MidiSetupListener* const listener)
     {
+        ScopedLock lock(mutex);
         listeners.addIfNotAlreadyThere(listener);
     }
 
     void removeListener(MidiSetupListener* const listener)
     {
+        ScopedLock lock(mutex);
         listeners.removeFirstMatchingValue(listener);
     }
 
     void midiDevicesChanged()
     {
+        ScopedLock lock(mutex);
         for (auto& listener : listeners)
         {
             listener->midiDevicesChanged();
         }
     }
 
 private:
+    CriticalSection mutex;

```

Comment:
Is `CriticalSection` re-entrant (like `std::recursive_mutex`)? Otherwise I see deadlock potential here...